### PR TITLE
Migration to osx-arm64

### PR DESCRIPTION
**Pull request by anjos**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Closes #28 (adds selective testing as per discussion on https://github.com/conda-forge/conda-forge.github.io/issues/1781).

### CODE REVIEW

**Comment by anjos** [comment]
@xhochy: I'm writing directly here, because I don't want to pollute (https://github.com/conda-forge/conda-forge.github.io/issues/1781).

> We avoid native testing for all osx-arm64 feedstocks and only run the tests for the feedstock that are just existence tests (i.e. using `test -f`)

Actually, there is more than just testing on this one - [the build executes the Lua interpreter](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=885930&view=logs&jobId=ced5d8de-8227-5f3f-33a1-45cf1592c45a&j=ced5d8de-8227-5f3f-33a1-45cf1592c45a&t=f7bcb349-6dca-53ba-d88d-1d9963e119ba) to get build information.  That cannot possibly work unless on a native build machine.  Any other ideas?

**Comment by xhochy** [comment]
> Any other ideas?

Yes. You should install `lua` also in build in the cross-setting and supply that as the lua interpreter for the configure script. The following line indicates that you should be able to do it:

```
Lua interpreter found: $PREFIX/bin/lua
```

**Comment by anjos** [comment]
> > Any other ideas?
> 
> Yes. You should install `lua` also in build in the cross-setting and supply that as the lua interpreter for the configure script. 

Thanks for this input!  Could you please be more explicit in what I should change to get this working properly?  From [this description of @isuruf's post](https://conda-forge.org/blog/2020/10/29/macos-arm64/), this is not clear in this particular case.

**Comment by xhochy** [comment]
You need to supply to the `./configure` call, e.g.

```
./configure --some-args=$BUILD_PREFIX/bin/lua
```

**Comment by anjos** [comment]
Thanks @xhochy - I think I found a way to make it work re-reading @isuruf's guide.

@jerowe, @izahn - this is ready for your review.  Hopefully, it can be merged before #30.

**Comment by anjos** [comment]
@xhochy :  pr #30 is ready for a bit more than a week without reply from current maintainers.  Should I add myself as maintainer here?  Can you merge it then? Thanks!

**Comment by anjos** [comment]
@conda-forge/lua - could any one of you please check and accept this PR?  It's been only 4 days, however #30 is there for about 10 w/o input.  I also added myself as maintainer to help.  Thanks in advance!

**Comment by anjos** [comment]
@conda-forge/core could you please review this PR and accept it please?  It's been [7 days w/o a reaction from assigned maintainers](https://conda-forge.org/docs/maintainer/maintainer_faq/#a-feedstock-has-been-abandoned-and-i-would-like-to-take-over-maintenance).  I also added myself as maintainer to help. Thanks in advance!

File: recipe/meta.yaml
Comment (xhochy) [inline recipe/meta.yaml]: You should instead add `test: native_and_emulated` to `conda-forge.yaml` *and rerender*.
```suggestion
    - luarocks config
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by xhochy** [comment]
Added one comment, happy to merge it if that is addressed.

File: recipe/meta.yaml
Comment (anjos) [reply recipe/meta.yaml]: `conda-forge.yaml` already contains the suggested instructions and was rendered as such.  I'll try applying your suggestion and see if the build passes.

**Comment by anjos** [comment]
@conda-forge-admin, please restart ci

**Comment by anjos** [comment]
@xhochy: done as requested and builds are passing!

**Review by xhochy — APPROVED** [review]

**Comment by anjos** [comment]
Thank you, @xhochy 🎉

**Pull request merged**
